.github/PULL_REQUEST_TEMPLATE/back-end-pr-template-systems-intelligence.md

```diff
@@ -33,12 +33,13 @@
 <!--
     Migration Script: <link to migration script file>
     Description: <a description of the specific schema or data changes made by this migration script>
-
-    Keep this tag - the data enablement team will be notified via email of any changes: @nicheinc/data-enablement
 -->
 
 <!-- Does this PR add PII to a new table? Consult Anonymizing User Data: https://bookstack.niche.team/books/back-end-patterns-practices/page/anonymizing-user-data -->
 
+<!-- Does this migration affect tables in Snowflake? If yes, submit a Data Team Request to have the schema migration reflected in Snowflake.
+https://form.asana.com/?k=uNnfTyZZVsxifMCOIPdwiw&d=684757491145461 -->
```

Comment:
❓️/💭 Is the expectation that not all schema migrations that add/remove tables need to be reflected in Snowflake - and that it's up to the dev to decide on a per-migration basis? Whether we're tagging a GH team or opening an Asana ticket, IMO it should still be the Data Team's call, and in that case, I think it might make more sense for this callout to be grouped with the schema migration comment above. 🤔

Reply:
Yeah so schema changes don't necessarily need to happen in Snowflake if the affected columns/schemas only serve backend services and not Snowflake stakeholders. 

I think we could get more eyes on this from backend and data leadership so they can decide what's needed and what's not

Reply:
Is there a strong reason not to say "all schema migrations should be reflected in snowflake, even if they're trivial"? I feel like I'm missing something here.

It feels like we're setting ourselves up for future problems if we decide there are migrations that don't need to be reflected in snowflake because:

1. We can always be wrong when we make this decision - regardless of who is involved or how many people or automated systems we use
2. Adding more layers of review will slow down velocity, either of updates to snowflake (best case) or actual deployments (will make everyone sad).

Reply:
Yeah so the schema migration process in snowflake often times involves re-snapshotting a table to backfill data from added columns. This re-snapshotting process needs to happen manually if the column data is populated with default values, because the Postgres WAL doesn't identify those as DML, and Streamkap won't pick these up as new events. It often takes some time to process/takes snowflake compute resources that cost money. 

So depending on how often schema migrations occur, @jack-mcquown said it can happen a lot for certain services. It may be inefficient to add all schema migrations to Snowflake. Unless we add some note in backend schema migrations to use DML to update new columns rather than using default values.